relayer/provider/matcher.go

```diff
@@ -118,3 +154,47 @@ func isMatchingTendermintClient(a, b tmclient.ClientState) bool {
 func isMatchingTendermintConsensusState(a, b *tmclient.ConsensusState) bool {
 	return reflect.DeepEqual(*a, *b)
 }
+
+// checkTendermintMisbehaviour checks that a proposed consensus state, used to update a tendermint light client,
+// matches the trusted consensus state from the counterparty chain. If there is no cached trusted header then
+// it will be queried from the counterparty. If the consensus states for the proposed header and the trusted header
+// match then both returned values will be nil.
+func checkTendermintMisbehaviour(
+	ctx context.Context,
+	clientID string,
+	proposedHeader *tmclient.Header,
+	cachedHeader IBCHeader,
+	counterparty ChainProvider,
+) (ibcexported.ClientMessage, error) {
+	var (
+		trustedHeader *tmclient.Header
+		err           error
+		ok            bool
+	)
+
+	if cachedHeader == nil {
+		header, err := counterparty.QueryHeaderAtHeight(ctx, proposedHeader.Header.Height)
```

Comment:
Is there a reason this needs to use `QueryHeaderAtHeight` instead of `QueryIBCHeader`?

Reply:
Good catch, this is something I was using before we added the function to `TendermintIBCHeader` to return the actual type we needed from ibc-go. I'll adjust and remove this